docs/smart-contracts/guides/transactions/simulateTransaction-Deep-Dive.mdx

```diff
@@ -0,0 +1,531 @@
+  // Construct the transaction builder with the necessary parameters
+  const restoreTx = new TransactionBuilder(account, {
+    ...txParams.txBuilderOptions,
+    fee: restorePreamble.minResourceFee, // Update fee based on the restoration requirement
+  })
+    .setSorobanData(restorePreamble.transactionData.build()) // Set Soroban Data from the simulation
+    .addOperation(Operation.restoreFootprint({})) // Add the RestoreFootprint operation
+    .build(); // Build the transaction
+
+  const simulation: SorobanRpc.Api.SimulateTransactionResponse =
+    await config.rpc.simulateTransaction(restoreTx);
+  const assembledTx = SorobanRpc.assembleTransaction(
+    restoreTx,
+    simulation,
+  ).build();
+
+  const signedTx = new Transaction(
+    await txParams.signerFunction(assembledTx.toXDR()),
+    config.passphrase,
+  );
+
+  console.log("Submitting restoration transaction");
+
+  try {
+    // Submit the transaction to the network
+    const response = await config.rpc.sendTransaction(signedTx);
+    console.log(
+      "Restoration transaction submitted successfully:",
+      response.hash,
+    );
+  } catch (error) {
+    console.error("Failed to submit restoration transaction:", error);
+    throw new Error("Restoration transaction failed");
+  }
+}
+
+import {
+  Account,
+  Keypair,
+  Operation,
+  SorobanDataBuilder,
+  SorobanRpc,
+  TimeoutInfinite,
+  Transaction,
+  TransactionBuilder,
+  scValToNative,
+  xdr,
+} from "@stellar/stellar-sdk";
+```
+
+## Fees and Resource Usage
+
+Soroban smart contracts on Stellar use a multidimensional resource fee model, charging fees for several resource types:
+
+1. **CPU Instructions**: Number of CPU instructions the transaction uses.
+2. **Ledger Entry Accesses**: Reading or writing any single ledger entry.
+3. **Ledger I/O**: Number of bytes read from or written to the ledger.
+4. **Transaction Size**: Size of the transaction submitted to the network in bytes.
+5. **Events & Return Value Size**: Size of the events produced by the contract and the return value of the top-level contract function.
+6. **Ledger Space Rent**: Payment for ledger entry TTL extensions and rent payments for increasing ledger entry size.
+
+Fees are calculated based on the resource consumption declared in the
+
+transaction. Refundable fees are charged before execution and refunded based on actual usage, while non-refundable fees are calculated from CPU instructions, read bytes, write bytes, and transaction size.
+
+## Preflight Error Handling
+
+The preflight mechanism provides an estimation of CPU and memory consumption in a transaction. It helps identify potential errors and resource limitations before actual submission. Errors returned by the host are propagated through RPC and do not cover network errors or errors in RPC itself.
+
+### Common Preflight Errors
+
+- **HostError(Budget, LimitExceeded)**: Network-defined resource limit exceeded. Optimize the contract to consume fewer resources.
```

Comment:
suggestion: formatting to have better punctuation flow (applies to all), i.e. `Cause: course of action.`
```suggestion
- **HostError(Budget, LimitExceeded)**: A network-defined resource limit was exceeded: optimize the contract to consume fewer resources.
```

Reply:
This!